backend/infrahub/core/schema/schema_branch.py

```diff
@@ -1740,3 +1765,202 @@ def generate_profile_from_node(self, node: NodeSchema) -> ProfileSchema:
+            node = self.get(name=node_name, duplicate=False)
+
+            if (
+                node.namespace in RESTRICTED_NAMESPACES
+                or not node.generate_template
+                or node.state == HashableModelState.ABSENT
+            ):
+                continue
+
+            template_rel_settings: dict[str, Any] = {
+                "name": "object_template",
+                "identifier": "node__objecttemplate",
+                "peer": self._get_object_template_kind(node.kind),
+                "kind": RelationshipKind.ATTRIBUTE,
+                "cardinality": RelationshipCardinality.ONE,
+                "branch": BranchSupportType.AWARE,
+                "order_weight": 1,
+            }
+
+            # Add relationship between node and template
+            if "object_template" not in node.relationship_names:
+                node_schema = self.get(name=node_name, duplicate=True)
+
+                node_schema.relationships.append(RelationshipSchema(**template_rel_settings))
+                self.set(name=node_name, schema=node_schema)
+            else:
+                has_changes: bool = False
+                rel_template = node.get_relationship(name="object_template")
+                for name, value in template_rel_settings.items():
+                    if getattr(rel_template, name) != value:
+                        has_changes = True
+
+                if not has_changes:
+                    continue
+
+                node_schema = self.get(name=node_name, duplicate=True)
+                rel_template = node_schema.get_relationship(name="object_template")
+                for name, value in template_rel_settings.items():
+                    if getattr(rel_template, name) != value:
+                        setattr(rel_template, name, value)
+
+                self.set(name=node_name, schema=node_schema)
+
+    def add_relationships_to_template(self, node: NodeSchema) -> None:
+        template_schema = self.get(name=self._get_object_template_kind(node_kind=node.kind), duplicate=False)
+
+        for relationship in node.relationships:
+            if relationship.peer in [
+                InfrahubKind.GENERICGROUP,
+                InfrahubKind.PROFILE,
+            ] or relationship.kind not in [RelationshipKind.COMPONENT, RelationshipKind.PARENT]:
+                continue
+
+            rel_template_peer = self._get_object_template_kind(node_kind=relationship.peer)
+            template_schema.relationships.append(
+                RelationshipSchema(
+                    name=relationship.name,
+                    peer=rel_template_peer,
+                    kind=relationship.kind,
+                    optional=relationship.kind == RelationshipKind.COMPONENT,
+                    cardinality=relationship.cardinality,
+                    branch=relationship.branch,
+                    identifier="__".join(sorted([template_schema.kind.lower(), rel_template_peer.lower()])),
```

Comment:
It would help to have a single function building this identifier to use both here and within `generate_identifiers`